test/integration/buyers/accounts_controller_test.rb

```diff
@@ -316,6 +316,13 @@ def setup
       assert_xpath( './/div[@id="applications_widget"]//table[@class="list"]//tr', 2)
       refute_xpath( './/div[@id="applications_widget"]//table[@class="list"]//tr', /plan/i )
     end
+
+    test 'suspend button is not displayed when account is deleted or marked for deletion' do
+      ThreeScale.config.stubs(onpremises: false)
+      delete admin_buyers_account_path(@provider)
+      assert_select %(td a.button-to.action.suspend), false
```

Comment:
I think this test could cover at least two other cases:
1. account NOT marked for deletion and NOT suspended
2. account NOT marked for deletion but already suspended

Reply:
Is there any method I can use to suspend or delete the account?
Something like `@provider.suspend`, `@provider.resume` ?
Or it's done by post request?

Reply:
I updated the test, please take a look